x-pack/plugins/observability/public/pages/rule_details/components/actions.tsx

```diff
@@ -15,45 +15,43 @@ import {
   EuiLoadingSpinner,
 } from '@elastic/eui';
 import { intersectionBy } from 'lodash';
+import { suspendedComponentWithProps } from '@kbn/triggers-actions-ui-plugin/public';
 import { ActionsProps } from '../types';
 import { useFetchRuleActions } from '../../../hooks/use_fetch_rule_actions';
 import { useKibana } from '../../../utils/kibana_react';
 
-interface MapActionTypeIcon {
-  [key: string]: string | IconType;
-}
-const mapActionTypeIcon: MapActionTypeIcon = {
-  /* TODO:  Add the rest of the application logs (SVGs ones) */
-  '.server-log': 'logsApp',
-  '.email': 'email',
-  '.pagerduty': 'apps',
-  '.index': 'indexOpen',
-  '.slack': 'logoSlack',
-  '.webhook': 'logoWebhook',
-};
-export function Actions({ ruleActions }: ActionsProps) {
+export function Actions({ ruleActions, actionTypeRegistry }: ActionsProps) {
   const {
     http,
     notifications: { toasts },
   } = useKibana().services;
   const { isLoadingActions, allActions, errorActions } = useFetchRuleActions({ http });
   if (ruleActions && ruleActions.length <= 0) return <EuiText size="s">0</EuiText>;
+
+  function getActionIconClass(actionGroupId?: string): IconType | undefined {
+    const actionGroup = actionTypeRegistry.list().find((group) => group.id === actionGroupId);
+    return typeof actionGroup?.iconClass === 'string'
+      ? actionGroup?.iconClass
+      : suspendedComponentWithProps(actionGroup?.iconClass as React.ComponentType);
+  }
   const actions = intersectionBy(allActions, ruleActions, 'actionTypeId');
   if (isLoadingActions) return <EuiLoadingSpinner size="s" />;
   return (
     <EuiFlexGroup direction="column">
-      {actions.map((action) => (
-        <>
+      {actions.map(({ actionTypeId, name }) => (
+        <React.Fragment key={actionTypeId}>
           <EuiFlexGroup alignItems="baseline">
```

Comment:
```suggestion
          <EuiFlexGroup alignItems="center">
            <EuiFlexItem
              style={{
                position: 'relative',
                width: '26px',
                height: '26px',
              }}
              grow={false}
            >
              <EuiIcon
                style={{
                  position: 'absolute',
                  height: '100%',
                  width: '100%',
                  left: 0,
                  top: 0,
                }}
                size="m"
                type={getActionIconClass(actionTypeId) ?? 'apps'}
              />
            </EuiFlexItem>
```

@mgiota, It's not perfect, but it seems a bit better. feel free to add it if you think so
<img width="301" alt="Screenshot 2022-05-19 at 14 33 46" src="https://user-images.githubusercontent.com/6838659/169294318-b23966da-cc2b-4b8d-b649-bfca08b5ea29.png">

Reply:
@fkanout Great,  I back to this PR and I am gad to see it is already look a bit better. I can take it from there and do a bit of refactoring regarding using `import { euiStyled } from '@kbn/kibana-react-plugin/common';` to not do the styling in place. 

@katrin-freihofner Do the icons look big? I can play with the dimensions and change the width and height of `26px`

Reply:
@katrin-freihofner You can read my comment above as well https://github.com/elastic/kibana/pull/132026#issuecomment-1131590593

Reply:
Good news, I played a bit more with the xmatters svg file and I fixed. Step 2 and custom dimensions are not needed anymore, since xmatters behaves well now with the predefined `size="m"` :-)

<img width="340" alt="Screenshot 2022-05-19 at 22 37 00" src="https://user-images.githubusercontent.com/2852703/169400281-3920f4b0-2b6b-4abc-be93-e9c0629faed4.png">

Reply:
Oops I didn't try with smaller icons like IBM, so I assume these will look a bit small. I'll give it a try and if needed I will add custom width and height so that all look the same.

Reply:
@mgiota, great work! 👏🏻

Reply:
@fkanout Small issues surfaced with this PR, good that it looks much better now! Your approval is still needed for this PR.